omnigibson/object_states/temperature.py

```diff
@@ -41,17 +44,30 @@ def _set_value(self, new_value):
         return True
 
     def _update(self):
+        # Avoid circular import
+        from omnigibson.object_states.on_fire import OnFire
+
         # Start at the current temperature.
         new_temperature = self.value
 
         # Find all heat source objects.
         affected_by_heat_source = False
-        for obj2 in self._simulator.scene.get_objects_with_state(HeatSourceOrSink):
-            # Obtain heat source position.
-            heat_source = obj2.states[HeatSourceOrSink]
-            heat_source_state, heat_source_position = heat_source.get_value()
+        heat_sources = og.sim.scene.get_objects_with_state(HeatSourceOrSink) | \
+                       og.sim.scene.get_objects_with_state(OnFire)
+        for obj2 in heat_sources:
+            # Only external heat sources will affect the temperature.
+            if obj2 == self.obj:
+                continue
+
+            heat_source = obj2.states[OnFire] if OnFire in obj2.states else obj2.states[HeatSourceOrSink]
```

Comment:
Does this imply that objects will either be only OnFire or HeatSourceOrSink?

Reply:
Added a sanity check.